Race in unit test: Egress Service Operations

```
2022-09-09T14:39:13.8316190Z [32m•[0m
2022-09-09T14:39:13.8316803Z JUnit path was configured: /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/_artifacts/junit-pkg_ovn.xml
2022-09-09T14:39:13.8365251Z 
2022-09-09T14:39:13.8365824Z JUnit report was created: /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/_artifacts/junit-pkg_ovn.xml
2022-09-09T14:39:13.8366822Z 
2022-09-09T14:39:13.8367432Z [1m[32mRan 187 of 187 Specs in 372.627 seconds[0m
2022-09-09T14:39:13.8368723Z [1m[32mSUCCESS![0m -- [32m[1m187 Passed[0m | [91m[1m0 Failed[0m | [33m[1m0 Pending[0m | [36m[1m0 Skipped[0m
2022-09-09T14:39:13.8369519Z     testing.go:1312: race detected during execution of test
2022-09-09T14:39:13.8371679Z --- FAIL: TestClusterNode (372.86s)
2022-09-09T14:39:13.8374793Z === CONT  
2022-09-09T14:39:13.8375127Z     testing.go:1312: race detected during execution of test
2022-09-09T14:39:13.8375611Z FAIL
```

--- a/go-controller/pkg/metrics/master.go
+++ b/go-controller/pkg/metrics/master.go
@@ -843,7 +843,12 @@ type ConfigDurationRecorder struct {
 // global variable is needed because this functionality is accessed in many functions
 var cdr *ConfigDurationRecorder
 
+// lock for accessing the cdr global variable
+var cdrMutex sync.Mutex
+
 func GetConfigDurationRecorder() *ConfigDurationRecorder {
+	cdrMutex.Lock()
+	defer cdrMutex.Unlock()
 	if cdr == nil {
 		cdr = &ConfigDurationRecorder{}
 	}

